File: leet/0021-merge-two-sorted-lists/merge-two-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {  
    dummy := &ListNode{Val: -1}
    last := dummy
    
    for l1 != nil || l2 != nil {
        if l1 == nil {
            last.Next = l2
            break;
        }
        if l2 == nil {
            last.Next = l1
            break;
        }
        
        if l1.Val < l2.Val{
            last.Next = l1
            last = last.Next
            l1 = l1.Next
        }else{
            last.Next = l2
            last = last.Next
            l2 = l2.Next
        }
    }
    
    return dummy.Next
}

File: leet/0021-merge-two-sorted-lists/merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
    let dummy = new ListNode()
    let last = dummy
    
    while(l1 && l2){
        if (l1.val < l2.val) {
            last.next = l1
            l1 = l1.next
        }else{
            last.next = l2
            l2 = l2.next
        }
        last = last.next
    }
    
    last.next = l1 ? l1 : l2
    return dummy.next
};
